gapic/schema/api.py

```diff
@@ -493,6 +496,57 @@ def get_custom_operation_service(self, method: "wrappers.Method") -> "wrappers.S
 
         return op_serv
 
+    @property
+    def mixin_api_methods(self) -> Dict[str, MethodDescriptorProto]:
+        methods: Dict[str, MethodDescriptorProto] = {}
+        if self.has_location_mixin():
+            methods = {**methods, **
+                self._get_methods_from_service(locations_pb2)}
+        if not self._has_iam_overrides() and self.has_iam_mixin():
+            methods = {**methods, **
+                self._get_methods_from_service(iam_policy_pb2)}
+        # For LRO, expose operations client instead.
+        return methods
+
+    def has_location_mixin(self) -> bool:
+        return len(list(filter(lambda api: api.name == "google.cloud.location.Locations", self.service_yaml_config.apis))) > 0
+
+    def has_iam_mixin(self) -> bool:
+        return len(list(filter(lambda api: api.name == "google.iam.v1.IAMPolicy", self.service_yaml_config.apis))) > 0
+
+    def has_operations_mixin(self) -> bool:
+        return len(list(filter(lambda api: api.name == "google.longrunning.Operations", self.service_yaml_config.apis))) > 0
+
+    def _has_iam_overrides(self) -> bool:
+        if not self.has_iam_mixin():
+            return False
+        iam_mixin_methods: Dict[str, MethodDescriptorProto] = self._get_methods_from_service(
+            iam_policy_pb2)
+        for (_, s) in self.services.items():
+            for m_name in iam_mixin_methods:
+                if m_name in s.methods:
```

Comment:
what is time complexity of search in `s.methods`? Is it O(1)? Just trying to avoid quadratic performance here.

Reply:
`s.methods` is a map, so O(1).